From 4eaedba0f3dd3bfdc5659f509013667b33507e44 Mon Sep 17 00:00:00 2001
From: Allain Legacy <allain.legacy@windriver.com>
Date: Sat, 14 Feb 2015 23:11:19 +0000
Subject: [PATCH 1/3] mk: updated makefiles to support new compilers

The invocation of the gcc preprocessor to generate the DPDK config file is broken on newer
compilers when the config files have #include statements.  This change prevents the
preprocessor from prepending C-style copyright headers to the file.
---
 mk/rte.sdkconfig.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mk/rte.sdkconfig.mk b/mk/rte.sdkconfig.mk
index ace65b0..8345d14 100644
--- a/mk/rte.sdkconfig.mk
+++ b/mk/rte.sdkconfig.mk
@@ -55,7 +55,7 @@ $(RTE_OUTPUT)/.config: $(RTE_CONFIG_TEMPLATE) FORCE
 	@[ -d $(RTE_OUTPUT) ] || mkdir -p $(RTE_OUTPUT)
 	$(Q)if [ "$(RTE_CONFIG_TEMPLATE)" != "" -a -f "$(RTE_CONFIG_TEMPLATE)" ]; then \
 		if grep -q '#include' $(RTE_CONFIG_TEMPLATE) ; then \
-			$(CPP) -undef -C -P -x assembler-with-cpp -fdirectives-only \
+			$(CPP) -undef -C -P -x assembler-with-cpp -fdirectives-only -ffreestanding \
 			-o $(RTE_OUTPUT)/.config_tmp $(RTE_CONFIG_TEMPLATE) ; \
 		else \
 			cp $(RTE_CONFIG_TEMPLATE) $(RTE_OUTPUT)/.config_tmp ; \
-- 
1.9.1

